### PR TITLE
add qmljs

### PR DESCRIPTION
**Pull request by Decodetalkers**
Add the treesitter of qmljs

### CODE REVIEW

**Comment by Decodetalkers** [comment]
I add qmljs support from the repo of yuja, and simple highlight.. and the highlight maybe need update for better experience..

**Comment by Decodetalkers** [comment]
image

know it is look like this...Still a lot of place white

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -43,7 +43,15 @@ local list = setmetatable({}, {
     filetype_to_parsername[parserconfig.filetype or parsername] = parsername
   end,
 })
-
+list.qmljs = {
+  install_info = {
+    url = "https://github.com/yuja/tree-sitter-qmljs",
+    branch = {"main"},
```
[the remark this answers is not in the document]
Comment (Decodetalkers) [reply lua/nvim-treesitter/parsers.lua]: > should be a string

fixed

**Comment by Decodetalkers** [comment]
image

Finished

File: queries/qmljs/highlights.scm [hunk 2]
```diff
@@ -44,31 +44,205 @@
...
+  "try"
+  "typeof"
+  "var"
+  "void"
+  "while"
+  "with"
+  "yield"
 ] @keyword
+
+
```
Comment (kyazdani42) [inline queries/qmljs/highlights.scm:240]: should this extend the ecma queries ?

Comment (Decodetalkers) [reply queries/qmljs/highlights.scm:240]: Maybe should, but how? and what is it about?

Comment (Decodetalkers) [reply queries/qmljs/highlights.scm:242]: > should this extend the ecma queries ?

I inherts ecma, and get beautiful highlight, thanks

Comment (kyazdani42) [reply queries/qmljs/highlights.scm:284]: you should remove the duplicates from ecma queries then. All those keywords and matches might already be matched by the ecma query.

Comment (Decodetalkers) [reply queries/qmljs/highlights.scm]: > you should remove the duplicates from ecma queries then. All those keywords and matches might already be matched by the ecma query.

I have done

**Comment by Decodetalkers** [comment]
image
 Yes , it is better now

**Comment by Decodetalkers** [comment]
I see , I write a wrong branch

**Comment by Decodetalkers** [comment]
I am an idiot

**Comment by Decodetalkers** [comment]
Please have another check thanks

**Comment by Decodetalkers** [comment]
Sorry, maybe some problem..

**Comment by Decodetalkers** [comment]
Yes ,  please test it

**Comment by kyazdani42** [comment]
please fix the stylua issue (missing commas)

**Comment by Decodetalkers** [comment]
> please fix the stylua issue (missing commas)

And what about the other, why they still failled?

**Comment by kyazdani42** [comment]
seems like some of the tag you try to assign to captures don't exist

**Comment by Decodetalkers** [comment]
> 

The locals.scm is the file from the repo upstream, I maybe can try to delete it..

**Comment by Decodetalkers** [comment]
> seems like some of the tag you try to assign to captures don't exist

Now I delete the locals.scm, I think now it can pass

**Comment by Decodetalkers** [comment]
Another try?

**Comment by Decodetalkers** [comment]
I think I need help...

**Comment by kyazdani42** [comment]
some tags are incorrect. Please see the list [here](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#highlights)

File: queries/qmljs/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  name: [
+    (identifier) @property
+    (nested_identifier (identifier) @property)
+  ])
+
+;; locals query appears not working unless id: <ref> isn't a parameter.
+(ui_binding
+  name: (identifier) @property
+  (#eq? @property "id")
+  value: (expression_statement (identifier) @variable.parameter))
```
Comment (theHamsta) [inline queries/qmljs/highlights.scm:34]: `@parameter`

File: queries/qmljs/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+(ui_required
+  name: (identifier) @property)
+
+(ui_list_property_type
+  ["<" ">"] @punctuation.bracket)
+
+;;; Signals
+
+(ui_signal
+  name: (identifier) @function.signal)
```
Comment (theHamsta) [inline queries/qmljs/highlights.scm:48]: Does our CI allow this? It's good when it just checks that the prefix is in CONTRIBING.md

**Review by theHamsta — APPROVED** [review]
Thank you!

File: lua/nvim-treesitter/parsers.lua [hunk 5]
```diff
@@ -43,7 +43,15 @@ local list = setmetatable({}, {
     filetype_to_parsername[parserconfig.filetype or parsername] = parsername
   end,
 })
-
+list.qmljs = {
+  install_info = {
+    url = "https://github.com/yuja/tree-sitter-qmljs",
+    branch = "master",
+    files = { "src/parser.c", "src/scanner.c" },
```
[the remark this answers is not in the document]
Comment (kyazdani42) [reply lua/nvim-treesitter/parsers.lua:50]: @theHamsta not sure how to fuzz scanners. I think we should also fuzz the elm scanner which appears to segfault, i'd be happy to give it a shot :)

**Comment by Decodetalkers** [comment]
Now it passed on my mechine...I don't know if what I change is right..

**Comment by Decodetalkers** [comment]
by the way, the file should be state as qmljs, so that treesitter will work..Not qml but qmljs

**Review by kyazdani42 — APPROVED** [review]

**Comment by kyazdani42** [comment]
thanks for the contribution :)

**Pull request merged**